### PR TITLE
Refactoring The New Architecture Guide

### PR DESCRIPTION
**Pull request by cipolleschi**
## Goal

This PR contains a proposal for a new structure of [The New Architecture](https://reactnative.dev/docs/next/new-architecture-intro) section on the website. This section lives in the [Guides](https://reactnative.dev/docs/next/getting-started) tab, therefore it contains mostly tutorials and examples. For more technical information on specific topics, it points to the [Architecture](https://reactnative.dev/architecture/overview) tab. The proposed structure covers some topics that are currently missing, and it refactors part of the documentation that already exists.

You can look at the added files or you can look at the preview generated by the CI to see how it will be rendered. 
**The content of each section is a description of what should be written in other PRs that have this PR as their starting point!**

## Context

At the moment of writing, The New Architecture guide’s section is a migration guide from the old architecture to the new one. It does not explain the basic context needed to understand it (what the new architecture is and what are their main benefits) and there is no section that describes how the new architecture’s components and modules should look like. For the old architecture, these topics are presented in [Native Module](https://reactnative.dev/docs/next/native-modules-intro) and [Native Components](https://reactnative.dev/docs/next/native-components-android) sections.

Currently, the guide does not suggest any approach to create backward compatible code, and it does not show how to use the new app template.

### CODE REVIEW

**Comment by Simek** [comment]
> @Simek I added a few redirects in the file you suggested [here](https://github.com/facebook/react-native-website/pull/3029#issuecomment-1080816347). I'm not sure they were needed, I have run a quick search on when and where `next/` is used and it looks like that this PR does not add anything else to the one already there.

Thank you for the updates, but I think that I have failed to explain and analyze this properly earlier. 😅 The guides are versioned, `next` still exist in the path. I have check out the branch locally to examine all of the changes locally and it looks like my initial assumption was wrong there, which came from analyzing the `diff`. 

In this PR there are no moved pages nor changed IDs for old pages. This PR only adds new pages and restructurize the sidebar, which should not affect the URLs. I think that I was fixated earlier on added `the` in the path on new entries, for some reason I thought that you have change this for all the pages.

In other words this PR doesn't need any redirect entries, sorry for the confusion!

**Comment by cipolleschi** [comment]
@Simek Don't worry! I updated the PR, removing those changes! ;)

**Comment by cortinico** [comment]
> I would especially like to see the section on creating backwards-compatible code merged soon.

We're actively working on this and it's pretty high on our agenda. We'll merge the whole work as soon as we feel we reached a shippable state 👍  Thanks for asking though.

File: docs/the-new-architecture/backward-compatibility-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+id: backward-compatibility-troubleshooting
+title: Troubleshooting
+---
+
+This section contains solutions to common problems that can happen when developing a backward compatible module or component.
```
Comment (cortinico) [inline docs/the-new-architecture/backward-compatibility-troubleshooting.md:6]: @cipolleschi Do we want to remove this page for now?

**Comment by cipolleschi** [comment]
Ok, I have no idea why the deploy fails. I tried in different ways and locally, the links work. @Simek, whenever you have time, could you have a look at what I'm doing wrong here? 🙏

File: docs/the-new-architecture/backward-compatibility-turbomodules.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+
+This snippet uses the same `RCT_NEW_ARCH_ENABLED` flag used in the previous [section](#dependencies-ios). When this flag is not set, Xcode skips the lines within the `#ifdef` during compilation and it does not include them into the compiled binary.
+
+### Android
+
+As we can't use conditional compilation blocks on Android, we will define two different source sets. This will allow to create a backward compatible TurboModule with the proper source that is loaded and compiled depending on the used architecture.
+
+Therefore, you have to:
+
+1. Create a Native Module in the `src/oldarch` path. See [this guide](../native-modules-intro) to learn how to create a Native Module.
```
Comment (Simek) [inline docs/the-new-architecture/backward-compatibility-turbomodules.md:185]:
```suggestion
1. Create a Native Module in the `src/oldarch` path. See [this guide](../native-modules-intro) to learn how to create a Native Module.
```

This link did not work for me locally, probably the ID has changed since the first iteration on this PR. It's also one of the problematic links reported by Docusaurus.

File: docs/the-new-architecture/backward-compatibility-fabric-components.md [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+
+The above snippet uses the same `RCT_NEW_ARCH_ENABLED` flag used in the previous [section](#dependencies-ios). When this flag is not set, Xcode skips the lines within the `#ifdef` during compilation and it does not include them into the compiled binary. The compiled binary will have a the `RNMyComponentView.o` object but it will be an empty object.
+
+### Android
+
+As we can't use conditional compilation blocks on Android, we will define two different source sets. This will allow to create a backward compatible TurboModule with the proper source that is loaded and compiled depending on the used architecture.
+
+Therefore, you have to:
+
+1. Create a Native Component in the `src/oldarch` path. See [this guide](../native-components-intro) to learn how to create a Native Component.
```
Comment (Simek) [inline docs/the-new-architecture/backward-compatibility-fabric-components.md:236]:
```suggestion
1. Create a Native Component in the `src/oldarch` path. See [this guide](../native-components-intro) to learn how to create a Native Component.
```

This link doesn't work form me locally, and it's also reported by Docusaurus, however I'm not sure which page you want to reference here, there is no doc using `native-components-intro` id.

File: docs/the-new-architecture/pillars-fabric-components.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
+---
+id: pillars-fabric-components
+title: Fabric Components
+---
+
+This section contains a high-level introduction to Fabric components. It provides enough context to understand when a Fabric component is needed and how it roughly works.
+It points to the [Renderer](https://reactnative.dev/architecture/fabric-renderer) section of the [Architecture](https://reactnative.dev/architecture/overview) tab for a deep dive into the technical details.
```
Comment (Simek) [inline docs/the-new-architecture/pillars-fabric-components.md:7]:
```suggestion
It points to the [Renderer](/architecture/fabric-renderer) section of the [Architecture](/architecture/overview) tab for a deep dive into the technical details.
```

Also we can improve those links.

**Comment by Simek** [comment]
I feel also like, after the refactor, sidebar section became growth to the height, in which it would be nice to consider splitting it into smaller section for the UX (especially on the smaller screens). Maybe "migration" part could be extracted into top level category? It's fine if such split is not preferred, it is just the quick idea.

File: docs/the-new-architecture/pillars-fabric-components.md (hunk 4 above)
Comment (cipolleschi) [reply docs/the-new-architecture/pillars-fabric-components.md:7]: This will change as soon as we merge the actual guide! 👍
